examples/cifar10/test.cpp

```diff
@@ -22,16 +18,14 @@ void convert_image(const std::string &imagefilename,
                    int w,
                    int h,
                    vec_t &data) {
-  cv::Mat img = cv::imread(imagefilename);
-  if (img.data == nullptr) return;  // cannot open, or it's not an image
-  cv::Mat resized;
-  cv::resize(img, resized, cv::Size(w, h), .0, .0);
-  data.resize(w * h * resized.channels(), minv);
-  for (int c = 0; c < resized.channels(); ++c) {
-    for (int y = 0; y < resized.rows; ++y) {
-      for (int x = 0; x < resized.cols; ++x) {
+  image<> img(imagefilename, image_type::rgb);
+  image<> resized = resize_image(img, w, h);
+  data.resize(w * h * 3);
+  for (int c = 0; c < 3; ++c) {
+    for (int y = 0; y < w; ++y) {
+      for (int x = 0; x < h; ++x) {
         data[c * w * h + y * w + x] =
-          resized.data[y * resized.step[0] + x * resized.step[1] + c];
+          (maxv - minv) * (resized[y * w + x + c]) / 255.0 + minv;
```

Comment:
make sure that this indexing work as expected.

Notice that we are introducing a Tensor class which have proper accessors and you might want to you use it here. Probably Image API can be refactored with Tensor API too.
https://github.com/tiny-dnn/tiny-dnn/blob/master/tiny_dnn/core/framework/tensor.h